src/components/Dropdown.js

```diff
@@ -0,0 +1,270 @@
+    </svg>
+  )
+
+const StyledDropdown = styled(DropdownIcon)`
+  transform: ${p => !p.isSearchable && p.menuIsOpen && `rotate(90deg)`};
+  fill: ${p =>
+    p.isSearchable
+      ? `none`
+      : p.menuIsOpen || p.hasValue
+      ? p.theme.colors.dropdown.background
+      : p.theme.colors.dropdown.neutral};
+`
+
+const Control = props => {
+  const {
+    selectProps: { menuIsOpen, isValid, errorMessage },
+  } = props
+  return (
+    <>
+      <components.Control {...props}>{props.children}</components.Control>
+      {!isValid && !menuIsOpen && <StyledErrorMsg>{errorMessage}</StyledErrorMsg>}
+    </>
+  )
+}
+
+const MenuList = props => {
+  const {
+    selectProps: { isSearchable, inputValue, emptySearchDefaultOption, canCreateNewOption, value },
+    hasValue,
+    children,
+  } = props
+  return (
+    <components.MenuList {...props}>
+      {isSearchable && inputValue === '' ? (
+        <StyledUserMessage>{emptySearchDefaultOption}</StyledUserMessage>
+      ) : (
+        <>
+          {canCreateNewOption &&
+            hasValue &&
+            inputValue.toLowerCase() === value.label.toLowerCase() && (
+              <StyledUserMessage>Your input is the same as the previous value!</StyledUserMessage>
+            )}
+          {children}
+        </>
+      )}
+    </components.MenuList>
+  )
+}
+
+const DropdownIndicator = props => {
+  const {
+    selectProps: { isSearchable, menuIsOpen },
+    hasValue,
+  } = props
+  return (
+    <components.DropdownIndicator {...props}>
+      <StyledDropdown isSearchable={isSearchable} menuIsOpen={menuIsOpen} hasValue={hasValue} />
+    </components.DropdownIndicator>
+  )
+}
+
+const NoOptionsMessage = props => {
+  const {
+    selectProps: { canCreateNewOption },
+    children,
+  } = props
+  return (
+    <components.NoOptionsMessage {...props}>
+      {!canCreateNewOption && <StyledUserMessage>{children}</StyledUserMessage>}
+    </components.NoOptionsMessage>
+  )
+}
+
+const Dropdown = ({
+  options,
+  placeholder,
+  isSearchable,
+  onChange,
+  formatCreateLabel,
+  emptySearchDefaultOption,
+  noOptionsMessage,
+  isValid,
+  errorMessage,
+  theme,
+  canCreateNewOption,
+}) => {
+  // These props are used by react-select directly, the rest are custom props
+  const userProps = {
+    options,
+    placeholder,
+    isSearchable,
+    onChange,
+    formatCreateLabel,
+    noOptionsMessage,
+    theme,
+  }
+
+  return canCreateNewOption ? (
+    <StyledCreatableDropdown
+      classNamePrefix="react-select"
+      components={{
+        Control,
+        DropdownIndicator,
+        MenuList,
+        IndicatorSeparator: () => null,
+        NoOptionsMessage,
+      }}
+      maxMenuHeight={200}
+      emptySearchDefaultOption={emptySearchDefaultOption}
+      errorMessage={errorMessage}
+      canCreateNewOption
+      isValid={isValid}
+      {...userProps}
+    />
+  ) : (
+    <StyledNormalDropdown
```

Comment:
can we make the search function debounced (ie. uses a setTimeout to throttle searching)? would be a nice-to-have for the dropdowns that have a loooot of data. 

the searchbar in the FAQ page uses debounced search, feel free to borrow the logic there